lib/rubocop/cop/style/redundant_map_compact.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for usage of `select` or `reject`
+      # instead of `map { |value| simple_condition? ? value : next }.compact`.
+      #
+      # @example
+      #
+      #   # bad
+      #   array.map { |e| some_condition? ? e : next }.compact
+      #
+      #   # bad
+      #   array.map do |e|
+      #     if some_condition?
+      #       e
+      #     else
+      #       next
+      #     end
+      #   end.compact
+      #
+      #   # bad
+      #   array.map do |e|
+      #     next if some_condition?
+      #
+      #     e
+      #   end.compact
+      #
+      #   # bad
+      #   array.map do |e|
+      #     e if some_condition?
+      #   end.compact
+      #
+      #   # good
+      #   array.select { |e| some_condition? }
+      #
+      #   # good
+      #   array.reject { |e| some_condition? }
```

Comment:
It is compatible when the block has `unless` condition, but bad examples are only `if` condition. Maybe it's better to delete it.